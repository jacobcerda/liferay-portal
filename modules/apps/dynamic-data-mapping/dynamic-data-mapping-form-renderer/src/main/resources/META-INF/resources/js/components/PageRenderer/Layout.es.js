/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import React from 'react';

import {EVENT_TYPES, usePage} from '../../hooks/usePage.es';
import {Field} from '../Field/Field.es';
import * as DefaultVariant from './DefaultVariant.es';

export const Layout = ({components: Components = DefaultVariant, rows}) => {
	const {
		dispatch,
		store: {activePage, allowNestedFields, editable, pageIndex, spritemap},
	} = usePage();

	return (
		<Components.Rows
			activePage={activePage}
			editable={editable}
			pageIndex={pageIndex}
			rows={rows}
		>
			{({index: rowIndex, row}) => (
				<Components.Row key={rowIndex} row={row}>
					{({column, index}) => (
						<Components.Column
							activePage={activePage}
							allowNestedFields={allowNestedFields}
							column={column}
							editable={editable}
							index={index}
							key={index}
							pageIndex={pageIndex}
							rowIndex={rowIndex}
						>
							{(fieldProps) => (
								<Field
									{...fieldProps}
									activePage={activePage}
									editable={editable}
									key={
										fieldProps.field?.instanceId ??
										fieldProps.field.name
									}
									onBlur={(event) =>
										dispatch({
											payload: event,
											type: EVENT_TYPES.FIELD_BLUR,
										})
									}
									onChange={(event) =>
										dispatch({
											payload: event,
											type: EVENT_TYPES.FIELD_CHANGE,
										})
									}
									onFocus={(event) =>
										dispatch({
											payload: event,
											type: EVENT_TYPES.FIELD_FOCUS,
										})
									}
									pageIndex={pageIndex}
									spritemap={spritemap}
								/>
							)}
						</Components.Column>
					)}
				</Components.Row>
			)}
		</Components.Rows>
	);
};
