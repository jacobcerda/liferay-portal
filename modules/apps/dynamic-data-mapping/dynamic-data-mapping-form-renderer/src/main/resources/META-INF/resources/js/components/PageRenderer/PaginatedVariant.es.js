/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ClayButton from '@clayui/button';
import React from 'react';

import {Pagination} from '../Pagination.es';
import {PaginationControls} from '../PaginationControls.es';
import * as DefaultVariant from './DefaultVariant.es';

export const Container = ({
	activePage,
	children,
	pageIndex,
	pages,
	readOnly,
	showSubmitButton,
	strings,
	submitLabel,
}) => (
	<div className="ddm-form-page-container paginated">
		<DefaultVariant.Container activePage={activePage} pageIndex={pageIndex}>
			{children}
		</DefaultVariant.Container>

		{pageIndex === activePage && (
			<>
				{pages.length > 0 && (
					<>
						<Pagination activePage={activePage} pages={pages} />
						<PaginationControls
							activePage={activePage}
							readOnly={readOnly}
							showSubmitButton={showSubmitButton}
							strings={strings}
							submitLabel={submitLabel}
							total={pages.length}
						/>
					</>
				)}

				{!pages.length && showSubmitButton && (
					<ClayButton
						className="float-right lfr-ddm-form-submit"
						type="submit"
					>
						{submitLabel}
					</ClayButton>
				)}
			</>
		)}
	</div>
);
