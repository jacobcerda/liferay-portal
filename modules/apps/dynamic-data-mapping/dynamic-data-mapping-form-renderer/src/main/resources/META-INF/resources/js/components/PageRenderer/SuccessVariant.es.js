/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {ClayButtonWithIcon} from '@clayui/button';
import {ClayDropDownWithItems} from '@clayui/drop-down';
import React, {useState} from 'react';

import {EVENT_TYPES, usePage} from '../../hooks/usePage.es';
import {setValue} from '../../util/i18n.es';

export const Container = ({children, pages, strings = {}}) => {
	const {
		dispatch,
		store: {editingLanguageId},
	} = usePage();

	return (
		<div className="page">
			<div className="ddm-page-success-layout sheet simple-page">
				<div className="form-builder-layout">
					<h5 className="pagination">
						{strings['success-page'] ??
							Liferay.Language.get('success-page')}
					</h5>

					{children}
				</div>
			</div>

			<div className="ddm-paginated-builder-dropdown">
				<ClayDropDownWithItems
					className="dropdown-action"
					items={[
						{
							label: Liferay.Language.get('remove-success-page'),
							onClick: () => {
								dispatch({
									payload: {
										body: {[editingLanguageId]: ''},
										enabled: false,
										title: {[editingLanguageId]: ''},
									},
									type: EVENT_TYPES.SUCCESS_CHANGED,
								});
								dispatch({
									payload: pages.length - 1,
									type: EVENT_TYPES.CHANGE_ACTIVE_PAGE,
								});
							},
						},
					]}
					trigger={
						<ClayButtonWithIcon
							displayType="unstyled"
							symbol="ellipsis-v"
							title={Liferay.Language.get('page-options')}
						/>
					}
				/>
			</div>
		</div>
	);
};

export const Page = ({page}) => {
	const {dispatch, store} = usePage();
	const {successPageSettings} = page;
	const {editingLanguageId} = store;

	const {initialBody, initialTitle} = {
		initialBody:
			(successPageSettings.body &&
				successPageSettings.body[editingLanguageId]) ||
			'',
		initialTitle:
			(successPageSettings.title &&
				successPageSettings.title[editingLanguageId]) ||
			'',
	};

	const [body, setBody] = useState(initialBody);
	const [title, setTitle] = useState(initialTitle);

	const onChange = (event, setting) => {
		dispatch({
			payload: setValue(
				successPageSettings,
				editingLanguageId,
				setting,
				event.target.value
			),
			type: EVENT_TYPES.SUCCESS_CHANGED,
		});
	};

	return (
		<div className="active ddm-form-page form-builder-success-page lfr-ddm-form-page">
			<input
				className="form-builder-page-header-title form-control p-0"
				maxLength="120"
				onChange={(event) => {
					setTitle(event.target.value);
					onChange(event, 'title');
				}}
				type="text"
				value={title}
			/>

			<input
				className="form-builder-page-header-description form-control p-0"
				maxLength="120"
				onChange={(event) => {
					setBody(event.target.value);
					onChange(event, 'body');
				}}
				type="text"
				value={body}
			/>
		</div>
	);
};
