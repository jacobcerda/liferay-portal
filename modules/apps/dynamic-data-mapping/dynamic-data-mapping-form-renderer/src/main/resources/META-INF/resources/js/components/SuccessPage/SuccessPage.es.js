/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import {setValue} from '../../util/i18n.es';
import templates from './SuccessPage.soy';

class SuccessPage extends Component {
	prepareStateForRender(state) {
		const {editingLanguageId, successPageSettings} = this;
		const {body, title} = successPageSettings;

		if (body[editingLanguageId] == '') {
			body[editingLanguageId] = Liferay.Language.get(
				'your-information-was-successfully-received-thank-you-for-filling-out-the-form'
			);
		}

		if (title[editingLanguageId] == '') {
			title[editingLanguageId] = Liferay.Language.get('thank-you');
		}

		return {
			...state,
			body: (body && body[editingLanguageId]) || '',
			enabled: true,
			title: (title && title[editingLanguageId]) || '',
		};
	}

	shouldUpdate(changes) {
		const {editingLanguageId} = changes;

		return (
			editingLanguageId &&
			editingLanguageId.newVal !== editingLanguageId.prevVal
		);
	}

	_handleSuccessPageUpdated(event) {
		const {dispatch, store} = this.context;
		const {editingLanguageId} = store.props;
		const {delegateTarget} = event;
		const {
			dataset: {setting},
			value,
		} = delegateTarget;
		const {successPageSettings} = this;

		dispatch(
			'successPageChanged',
			setValue(successPageSettings, editingLanguageId, setting, value)
		);
	}
}

SuccessPage.STATE = {
	editingLanguageId: Config.string(),

	strings: Config.object().value({
		'success-page': Liferay.Language.get('success-page'),
	}),

	/**
	 * @instance
	 * @memberof SuccessPage
	 * @type {?object}
	 */

	successPageSettings: Config.object().value({
		body: {},
		title: {},
	}),
};

Soy.register(SuccessPage, templates);

export default SuccessPage;
