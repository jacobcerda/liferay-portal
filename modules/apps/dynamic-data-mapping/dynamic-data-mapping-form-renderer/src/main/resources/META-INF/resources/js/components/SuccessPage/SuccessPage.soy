{namespace SuccessPage}

{template .render}
	{@param? _handleSuccessPageUpdated: any}
	{@param? body: string}
	{@param? strings: map<string, string>}
	{@param? title: string}

	{delcall SuccessPage.container}
		{param _handleSuccessPageUpdated: $_handleSuccessPageUpdated /}
		{param body: $body /}
		{param strings: $strings /}
		{param title: $title /}
	{/delcall}
{/template}

{deltemplate SuccessPage.container}
	{@param? _handleSuccessPageUpdated: any}
	{@param? body: string}
	{@param? strings: map<string, string>}
	{@param? title: string}

	<div class="sheet simple-page ddm-page-success-layout">
		<div class="form-builder-layout">
			<h5 class="pagination">{$strings != null ? $strings['success-page'] : ''}</h5>

			<div class="active ddm-form-page lfr-ddm-form-page form-builder-success-page">

				<input
					class="form-builder-page-header-title form-control p-0"
					maxlength="120"
					placeholder=""
					data-oninput="{$_handleSuccessPageUpdated}"
					data-setting="title"
					type="text"
					value="{$title}"
				/>

				<input
					class="form-builder-page-header-description form-control p-0"
					data-oninput="{$_handleSuccessPageUpdated}"
					data-setting="body"
					type="text"
					maxlength="120"
					placeholder=""
					value="{$body}"
				/>
			</div>
		</div>
	</div>
{/deltemplate}