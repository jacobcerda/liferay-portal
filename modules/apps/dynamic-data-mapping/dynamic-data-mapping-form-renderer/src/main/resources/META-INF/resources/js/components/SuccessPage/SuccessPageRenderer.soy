{namespace SuccessPageRenderer}

{deltemplate PageRenderer.container variant="'success'"}
	{@param activePage: number}
	{@param cancelLabel: string}
	{@param pageIndex: number}
	{@param pages: ?}
	{@param showCancelButton: bool}
	{@param showSubmitButton: bool}
	{@param spritemap: string}
	{@param submitLabel: string}
	{@param total: number}
	{@param? _handleFieldBlurred: any}
	{@param? _handleFieldClicked: any}
	{@param? _handleFieldEdited: any}
	{@param? editable: bool}
	{@param? editingLanguageId: string}
	{@param? empty: bool}
	{@param? page: ?}
	{@param? paginationMode: ?}
	{@param? readOnly: bool}
	{@param? redirectURL: string}
	{@param? strings: map<string, string>}

		{call SuccessPage.render}
			{param editingLanguageId: $editingLanguageId /}
			{param successPageSettings: $page?.successPageSettings /}
		{/call}
{/deltemplate}