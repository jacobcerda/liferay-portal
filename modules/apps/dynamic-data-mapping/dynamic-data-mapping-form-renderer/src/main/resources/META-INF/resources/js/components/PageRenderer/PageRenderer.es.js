/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import * as FormSupport from '../FormRenderer/FormSupport.es';

import './MultiPagesRenderer.soy';

import './PaginatedPageRenderer.soy';

import './SimplePageRenderer.soy';

import './TabbedPageRenderer.soy';

import './WizardPageRenderer.soy';

import 'clay-button';

import 'clay-dropdown';

import templates from './PageRenderer.soy';

import 'clay-modal';
import core from 'metal';
import Component from 'metal-component';
import dom from 'metal-dom';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import '../SuccessPage/SuccessPagePaginationItem.soy';

import '../SuccessPage/SuccessPageRenderer.soy';

import '../SuccessPage/SuccessPageWizardItem.soy';
import {setValue} from '../../util/i18n.es';

class PageRenderer extends Component {
	getPage(page) {
		const {editingLanguageId} = this;

		if (core.isObject(page.description)) {
			page = {
				...page,
				description: page.description[editingLanguageId],
			};
		}
		if (core.isObject(page.title)) {
			page = {
				...page,
				title: page.title[editingLanguageId],
			};
		}

		return page;
	}

	isEmptyPage({rows}) {
		let empty = false;

		if (!rows || !rows.length) {
			empty = true;
		}
		else {
			empty = !rows.some(({columns}) => {
				let hasFields = true;

				if (!columns) {
					hasFields = false;
				}
				else {
					hasFields = columns.some((column) => column.fields.length);
				}

				return hasFields;
			});
		}

		return empty;
	}

	prepareStateForRender(states) {
		if (states.view === 'fieldSets') {
			this._updateSuccessPage({
				body: '',
				enabled: false,
				title: '',
			});
		}

		return {
			...states,
			empty: this.isEmptyPage(states.page),
			page: this.getPage(states.page),
			pageSettingsItems: this._pageSettingsItems(states.pages),
			pageSuccessSettingsItems: this._pageSuccessSettingsItems(
				states.pages
			),
		};
	}

	_addPage(pageIndex) {
		const {dispatch} = this.context;

		dispatch('pageAdded', {
			pageIndex,
		});
	}

	_addSuccessPage() {
		const {dispatch} = this.context;
		const {pages} = this;

		this._updateSuccessPage({
			body: Liferay.Language.get(
				'your-information-was-successfully-received-thank-you-for-filling-out-the-form'
			),
			enabled: true,
			title: Liferay.Language.get('thank-you'),
		});

		dispatch('activePageUpdated', pages.length);
	}

	_deletePage() {
		const {dispatch} = this.context;

		dispatch('pageDeleted', this.pageIndex);
	}

	_deleteSuccessPage() {
		const {dispatch} = this.context;
		const {pages} = this;

		this._updateSuccessPage({
			enabled: false,
		});

		dispatch('activePageUpdated', pages.length - 1);
	}

	_resetPage(pageIndex) {
		const {dispatch} = this.context;

		dispatch('pageReset', {
			pageIndex,
		});
	}

	_handleAddPageClicked() {
		const {pageIndex} = this;

		this._addPage(pageIndex);
	}

	_handleAddSuccessPageClicked() {
		const {pageIndex} = this;

		this._addSuccessPage(pageIndex);
	}

	_handleDropdownItemClicked({data}) {
		const {pageIndex, successPageSettings} = this;
		const {settingsItem} = data.item;

		this.setState({
			dropdownExpanded: false,
		});

		if (settingsItem === 'reset-page') {
			this._resetPage(pageIndex);
		}
		else if (settingsItem === 'remove-page') {
			if (
				successPageSettings.enabled &&
				pageIndex == this.pages.length - 1
			) {
				this._deleteSuccessPage();
			}
			else {
				this._deletePage();
			}
		}
	}

	_handleFieldBlurred(event) {
		this.emit('fieldBlurred', event);
	}

	_handleFieldClicked(event) {
		const {delegateTarget} = event;
		const {fieldName} = delegateTarget.dataset;

		event.stopPropagation();

		this.emit('fieldClicked', {
			...FormSupport.getIndexes(dom.closest(delegateTarget, '.col-ddm')),
			fieldName,
			originalEvent: event,
		});
	}

	_handleFieldEdited(event) {
		this.emit('fieldEdited', event);
	}

	_handleFieldFocused(event) {
		this.emit('fieldFocused', event);
	}

	_pageSettingsItems(pages) {
		const {successPageSettings} = this;
		const pageSettingsItems = [
			{
				label: Liferay.Language.get('reset-page'),
				settingsItem: 'reset-page',
			},
		];

		if (
			(successPageSettings.enabled && pages.length > 2) ||
			(pages.length > 1 && !successPageSettings.enabled)
		) {
			pageSettingsItems.push({
				label: Liferay.Language.get('remove-page'),
				settingsItem: 'remove-page',
			});
		}

		return pageSettingsItems;
	}

	_pageSuccessSettingsItems() {
		const pageSuccessSettingsItems = [
			{
				label: Liferay.Language.get('remove-success-page'),
				settingsItem: 'remove-page',
			},
		];

		return pageSuccessSettingsItems;
	}

	_updateSuccessPage({body = '', enabled, title = ''}) {
		const {dispatch} = this.context;
		const {editingLanguageId} = this;
		const successPageSettings = {
			body: {},
			enabled,
			title: {},
		};

		setValue(successPageSettings, editingLanguageId, 'body', body);
		setValue(successPageSettings, editingLanguageId, 'title', title);

		dispatch('successPageChanged', successPageSettings);
	}
}

PageRenderer.STATE = {

	/**
	 * @instance
	 * @memberof FormPage
	 * @type {?number}
	 */

	activePage: Config.number().value(0),

	/**
	 * @instance
	 * @memberof FormPage
	 * @type {?number}
	 */

	dropdownExpanded: Config.bool().value(false),

	/**
	 * @instance
	 * @memberof FormPage
	 * @type {?boolean}
	 */
	editable: Config.bool().value(false),

	/**
	 * @default []
	 * @instance
	 * @memberof FormRenderer
	 * @type {?array<object>}
	 */

	page: Config.object(),

	/**
	 * @default 1
	 * @instance
	 * @memberof FormPage
	 * @type {?number}
	 */

	pageIndex: Config.number().value(0),

	/**
	 * @default 1
	 * @instance
	 * @memberof FormPage
	 * @type {?number}
	 */

	pageSettingsItems: Config.array(),

	/**
	 * @default 1
	 * @instance
	 * @memberof FormPage
	 * @type {?number}
	 */

	pageSuccessSettingsItems: Config.array(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof FormRenderer
	 * @type {!string}
	 */

	spritemap: Config.string().required(),

	strings: Config.object().value({
		'success-page': Liferay.Language.get('success-page'),
	}),

	successPageSettings: Config.shapeOf({
		body: Config.object(),
		enabled: Config.bool(),
		title: Config.object(),
	}).value({}),
};

Soy.register(PageRenderer, templates);

export default PageRenderer;
