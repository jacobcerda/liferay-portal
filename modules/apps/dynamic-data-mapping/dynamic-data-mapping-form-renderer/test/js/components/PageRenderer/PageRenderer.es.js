/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import '../../../../src/main/resources/META-INF/resources/js/components/SuccessPage/SuccessPage.es';
import PageRenderer from '../../../../src/main/resources/META-INF/resources/js/components/PageRenderer/PageRenderer.es';
import mockPagesEmpty from '../../__mock__/mockPageEmpty.es';
import mockSuccessPage from '../../__mock__/mockSuccessPage.es';
import withContextMock from '../../__mock__/withContextMock.es';

const spritemap = 'icons.svg';
let component;
let pagesEmpty;
let successPageSettings;
const PageRendererWithContextMock = withContextMock(PageRenderer);

const createPageRenderer = (props) => {
	return new PageRendererWithContextMock({
		editable: true,
		editingLanguageId: 'en_US',
		paginationMode: 'wizard',
		portletNamespace: 'portletNamespace',
		showSubmitButton: false,
		spritemap,
		submitLabel: 'Submit',
		successPageSettings,
		view: 'formBuilder',
		...props,
	});
};

describe('PageRenderer', () => {
	beforeEach(() => {
		pagesEmpty = JSON.parse(JSON.stringify(mockPagesEmpty));
		successPageSettings = JSON.parse(JSON.stringify(mockSuccessPage));

		jest.useFakeTimers();
	});

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('creates the defaut markup when success page is enabled', () => {
		pagesEmpty.push({contentRenderer: 'success'});

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[0],
			pageIndex: 0,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 2,
		});

		expect(component).toMatchSnapshot();
	});

	it('creates the defaut markup with button to add success page if success page is disabled', () => {
		successPageSettings.enabled = false;

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[0],
			pageIndex: 0,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 1,
		});

		expect(component).toMatchSnapshot();
		expect(component.refs.successPage).not.toBeUndefined();
	});

	it('shows the new page button on the bottom of each page', () => {
		pagesEmpty.push({contentRenderer: 'success'});

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[0],
			pageIndex: 0,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 2,
		});

		expect(component).toMatchSnapshot();
		expect(component.refs.newPage0).not.toBeUndefined();
	});

	it('propagates pageAdded event after option of remove a page be clicked', () => {
		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[0],
			pageIndex: 0,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 1,
		});

		component._handleAddPageClicked();

		expect(component.context.dispatch).toHaveBeenCalledWith('pageAdded', {
			pageIndex: 0,
		});
	});

	it('propagates pageDeleted event after option of remove a page be clicked', () => {
		pagesEmpty.push(pagesEmpty[0]);
		pagesEmpty.push({contentRenderer: 'success'});

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[1],
			pageIndex: 1,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 3,
		});

		const data = {
			item: {
				settingsItem: 'remove-page',
			},
		};

		component._handleDropdownItemClicked({data});

		expect(component.context.dispatch).toHaveBeenCalledWith(
			'pageDeleted',
			1
		);
	});

	it('propagates activePageUpdated event after option of remove success page be clicked', () => {
		pagesEmpty.push(pagesEmpty[0]);
		pagesEmpty.push({
			contentRenderer: 'success',
			rows: [],
			successPageSettings,
		});

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[2],
			pageIndex: 2,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 3,
		});

		const data = {
			item: {
				settingsItem: 'remove-page',
			},
		};

		component._handleDropdownItemClicked({data});

		expect(component.context.dispatch).toHaveBeenCalledWith(
			'activePageUpdated',
			2
		);
	});

	it('propagates pageReset event after option of remove reset page be clicked', () => {
		pagesEmpty.push(pagesEmpty[0]);
		pagesEmpty.push({contentRenderer: 'success'});

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[1],
			pageIndex: 1,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 3,
		});

		const data = {
			item: {
				settingsItem: 'reset-page',
			},
		};

		component._handleDropdownItemClicked({data});

		expect(component.context.dispatch).toHaveBeenCalledWith('pageReset', {
			pageIndex: 1,
		});
	});

	it('verify if kebab (three dots) menu of actions exist on Success Page', () => {
		pagesEmpty.push({contentRenderer: 'success'});

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[1],
			pageIndex: 1,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 2,
		});

		expect(component.refs.successPageActions1).not.toBeUndefined();
	});

	it('verify if kebab (three dots) menu of actions exist on a Page', () => {
		pagesEmpty.push({contentRenderer: 'success'});

		component = createPageRenderer({
			cancelLabel: 'cancel',
			page: pagesEmpty[0],
			pageIndex: 0,
			pages: pagesEmpty,
			showCancelButton: true,
			total: 2,
		});

		expect(component.refs.pageActions0).not.toBeUndefined();
	});
});
