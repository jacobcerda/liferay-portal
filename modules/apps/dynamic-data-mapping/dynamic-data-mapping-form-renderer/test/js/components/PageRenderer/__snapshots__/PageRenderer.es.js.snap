// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageRenderer creates the defaut markup when success page is enabled 1`] = `
<div class="page">
  <div class="sheet fade tab-pane show" role="tabpanel">
    <div class="form-builder-layout">
      <h5 class="pagination"> Page 1 of 1 </h5>
      <div class="active ddm-form-page lfr-ddm-form-page" data-ddm-page="0">
        <h3 class="lfr-ddm-form-page-title">Untitled name</h3>
        <div class="row">
          <div class="col col-ddm col-empty col-md-12 last-col lfr-initial-col mt-5 mb-4" data-ddm-field-column="0" data-ddm-field-page="0" data-ddm-field-row="0">
            <div class="ddm-empty-page ddm-target">
              <p class="ddm-empty-page-message">drag-fields-from-the-sidebar-to-compose-your-form</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ddm-paginated-builder-dropdown" data-ddm-page-actions="0">
    <div class="dropdown dropdown-action"><button aria-expanded="false" aria-haspopup="true" class="dropdown-toggle btn btn-unstyled" ref="triggerButton" title="actions" type="button"><svg class="lexicon-icon lexicon-icon-ellipsis-v" focusable="false" role="presentation">
          <use xlink:href="icons.svg#ellipsis-v"></use>
        </svg></button></div>
  </div>
  <div class="add-page-button-container">
    <div class="horizontal-line"></div><button class="btn add-page-button btn-sm btn-secondary" type="button">new-page</button>
    <div class="horizontal-line"></div>
  </div>
</div>
<div class="dropdown-menu" ref="menu">
  <ul class="list-unstyled">
    <li tabindex="-1"><button class="btn dropdown-item btn-sm btn-link" aria-label="reset-page" type="button">reset-page</button></li>
  </ul>
</div>
`;

exports[`PageRenderer creates the defaut markup with button to add success page if success page is disabled 1`] = `
<div class="page">
  <div class="sheet fade tab-pane show" role="tabpanel">
    <div class="form-builder-layout">
      <h5 class="pagination"> Page 1 of 1 </h5>
      <div class="active ddm-form-page lfr-ddm-form-page" data-ddm-page="0">
        <h3 class="lfr-ddm-form-page-title">Untitled name</h3>
        <div class="row">
          <div class="col col-ddm col-empty col-md-12 last-col lfr-initial-col mt-5 mb-4" data-ddm-field-column="0" data-ddm-field-page="0" data-ddm-field-row="0">
            <div class="ddm-empty-page ddm-target">
              <p class="ddm-empty-page-message">drag-fields-from-the-sidebar-to-compose-your-form</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ddm-paginated-builder-dropdown" data-ddm-page-actions="0">
    <div class="dropdown dropdown-action"><button aria-expanded="false" aria-haspopup="true" class="dropdown-toggle btn btn-unstyled" ref="triggerButton" title="actions" type="button"><svg class="lexicon-icon lexicon-icon-ellipsis-v" focusable="false" role="presentation">
          <use xlink:href="icons.svg#ellipsis-v"></use>
        </svg></button></div>
  </div>
  <div class="add-page-button-container">
    <div class="horizontal-line"></div><button class="btn add-page-button btn-sm btn-secondary" type="button">new-page</button>
    <div class="horizontal-line"></div>
  </div>
  <div class="add-page-button-container">
    <div class="horizontal-line"></div><button class="btn add-page-button btn-sm btn-secondary" type="button">add-success-page</button>
    <div class="horizontal-line"></div>
  </div>
</div>
<div class="dropdown-menu" ref="menu">
  <ul class="list-unstyled">
    <li tabindex="-1"><button class="btn dropdown-item btn-sm btn-link" aria-label="reset-page" type="button">reset-page</button></li>
  </ul>
</div>
`;

exports[`PageRenderer shows the new page button on the bottom of each page 1`] = `
<div class="page">
  <div class="sheet fade tab-pane show" role="tabpanel">
    <div class="form-builder-layout">
      <h5 class="pagination"> Page 1 of 1 </h5>
      <div class="active ddm-form-page lfr-ddm-form-page" data-ddm-page="0">
        <h3 class="lfr-ddm-form-page-title">Untitled name</h3>
        <div class="row">
          <div class="col col-ddm col-empty col-md-12 last-col lfr-initial-col mt-5 mb-4" data-ddm-field-column="0" data-ddm-field-page="0" data-ddm-field-row="0">
            <div class="ddm-empty-page ddm-target">
              <p class="ddm-empty-page-message">drag-fields-from-the-sidebar-to-compose-your-form</p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="ddm-paginated-builder-dropdown" data-ddm-page-actions="0">
    <div class="dropdown dropdown-action"><button aria-expanded="false" aria-haspopup="true" class="dropdown-toggle btn btn-unstyled" ref="triggerButton" title="actions" type="button"><svg class="lexicon-icon lexicon-icon-ellipsis-v" focusable="false" role="presentation">
          <use xlink:href="icons.svg#ellipsis-v"></use>
        </svg></button></div>
  </div>
  <div class="add-page-button-container">
    <div class="horizontal-line"></div><button class="btn add-page-button btn-sm btn-secondary" type="button">new-page</button>
    <div class="horizontal-line"></div>
  </div>
</div>
<div class="dropdown-menu" ref="menu">
  <ul class="list-unstyled">
    <li tabindex="-1"><button class="btn dropdown-item btn-sm btn-link" aria-label="reset-page" type="button">reset-page</button></li>
  </ul>
</div>
`;
